lib/XML/Sig.pm

```diff
@@ -428,11 +430,23 @@ sub sign {
         }
 
         # Add the Signature to the SignatureValue
-        my ($signature_value_node) = $xml->findnodes(
-            './dsig:Signature/dsig:SignatureValue', $signature_node);
+        my ($signature_value_node) = $xpath->findnodes(
+            '/dsig:Signature/dsig:SignatureValue', $signature_xml);
         $signature_value_node->removeChildNodes();
         $signature_value_node->appendText($signature);
 
+        my $set = $xpath->findnodes('dsig:Signature');
+
+        my $node;
```

Comment:
What happens when there isn't a node? This code assumes that the node exists?

Reply:
Base on the above we should likely dies because something would really have gone wrong.

my $node;
        if ($set->size) {
            $node = $set->get_node(1)->cloneNode( 1 );
        } else {
             die 'Unexpected death';
        }